src/Build.UnitTests/Graph/ProjectGraph_Tests.cs

```diff
@@ -769,6 +851,78 @@ public void GetTargetListsFiltersTargetsForOuterAndInnerBuilds()
             }
         }
 
+        [Fact]
+        public void GetTargetListsDoesNotUseTargetsMetadataOnInnerBuildsFromRootOuterBuilds()
```

Comment:
May I beg you for a diagram of this?

Reply:
![image](https://user-images.githubusercontent.com/2255729/58132013-10b41a80-7bd5-11e9-9c92-0b192b112416.png)